bayesian/utils.py

```diff
@@ -410,87 +411,123 @@ class CveByDateEcosystemUtils:
     """
 
     # Get CVEs information by date
-    cve_nodes_by_date_script_template = """\
-    g.V().has('modified_date', modified_date)\
-    .has('vertex_label','CVE').as('cve')\
-    .coalesce(\
-    inE('has_cve').outV().as('epv').select('cve','epv').by(valueMap()),\
-    select('cve').by(valueMap()))\
+    cve_epv_nodes_by_date_script_template = """\
+    g.V().has('cve_id', cve_id)\
+    .in("has_cve").valueMap()\
     """
 
     # Get CVEs by date & ecosystem
     cve_nodes_by_date_ecosystem_script_template = """\
     g.V().has('modified_date', modified_date)\
-    .has('vertex_label','CVE')\
-    .has('ecosystem',ecosystem).as('cve')\
-    .coalesce(\
-    inE('has_cve').outV().as('epv').select('cve','epv').by(valueMap()),\
-    select('cve').by(valueMap()))\
+    .has('cecosystem',ecosystem)\
+    .has('cve_sources',cve_sources)\
+    .valueMap()\
     """
 
-    def __init__(self, bydate, ecosystem=None):
-        """Constructor."""
-        self._bydate = bydate
-        self._ecosystem = ecosystem
+    cve_nodes_by_date_ecosystem_script_template_all = """\
+    g.V().has('modified_date', modified_date)\
+    .has('cecosystem',ecosystem)\
+    .valueMap()\
+    """
 
-    def get_cves_by_date(self):
-        """Call graph and get CVEs by date."""
-        script = self.cve_nodes_by_date_script_template
-        bindings = {'modified_date': self._bydate}
-        return self.get_cves(script, bindings)
+    def __init__(self, result_type, cve_id, cve_sources='all', bydate=None,
+                 ecosystem='all', date_range=7):
+        """Constructor."""
+        self._result_type = result_type
+        self._cve_id = cve_id
+        if cve_id is None:
+            self._bydate = bydate
+            self._ecosystem = ecosystem
+            self._cve_sources = cve_sources
+            self._date_range = date_range
 
     def get_cves_by_date_ecosystem(self):
         """Call graph and get CVEs by date and ecosystem."""
-        script = self.cve_nodes_by_date_ecosystem_script_template
-        self._ecosystem = self._ecosystem.lower()
-        bindings = {'modified_date': self._bydate, 'ecosystem': self._ecosystem}
-        return self.get_cves(script, bindings)
+        didx = pd.date_range(end=self._bydate, periods=self._date_range)
```

Comment:
getting the date range can be done with normal datetime functions. Please do not use pandas. That's a very heavy library to be included just for date range functions.